hudi-aws/src/main/java/org/apache/hudi/aws/transaction/lock/S3StorageLockClient.java

```diff
@@ -94,25 +92,13 @@ public S3StorageLockClient(String ownerId, String lockFileUri, Properties props)
 
   @VisibleForTesting
   S3StorageLockClient(String ownerId, String lockFileUri, Properties props, Functions.Function2<String, Properties, S3Client> s3ClientSupplier, Logger logger) {
-    try {
-      // This logic can likely be extended to other lock client implementations.
-      // Consider creating base class with utilities, incl error handling.
-      URI uri = new URI(lockFileUri);
-      this.bucketName = uri.getHost();
```

Comment:
For GCS, the bucket name is extracted by `String bucketName = uri.getAuthority()`, and the `parseBucketAndPath` uses `uri.getAuthority()`.  So should that work for S3 too (I remember `uri.getAuthority()` is less strict than `uri.getHost()`)?

Reply:
yes, we can do that here too